tools/autograd/templates/VariableType.cpp

```diff
@@ -461,10 +461,8 @@ Tensor & VariableType::resize_as_(Tensor & self, const Tensor & the_template) co
 
 Tensor VariableType::contiguous(const Tensor & self) const {
   unpack(self, "self", 0);
-  if (self.is_contiguous()) {
-    return self;
-  }
-  return self.clone();
+  auto result = self.is_contiguous() && !jit::tracer::isTracing() ? self : self.clone();
```

Comment:
Well tracing all contiguous calls as clones is not perfect because the traces can end up being *way* slower than the original code (if it's called a lot on contiguous tensors). The right solution would be to add the code which instantiates the Node for `aten::contiguous`.

Reply:
I'm working on this, let me ping you